File: regiondeployer/masterswitch.fs
module regiondeployer

open Microsoft.Azure.Management.Fluent
open Microsoft.Azure.Management.AppService.Fluent

[<Literal>]
let public masterfarmername = "chronictick"

let private getfirstmatch(logmessage:string -> unit) : IFunctionApp = 
    let functionapps = getfunctionapps(logmessage)
    let output = Seq.find(fun (currentapp:IFunctionApp) -> currentapp.Name.Contains(masterfarmername)) functionapps    
    output

let togglemasterfarmer (shouldrun:bool, logmessage:string -> unit) : unit =
    String.concat valueseparator ["toggling master farmers operational state to"; shouldrun.ToString()]
    |> logmessage

    let master = getfirstmatch(logmessage)
    
    if master = null then
        String.concat valueseparator [
            "no master farmer found. deploy that first. Its an azure function app whos name must contain"; 
            masterfarmername
        ]
        |> failwith 
        ()

    else
        match shouldrun with
        | true -> master.Start()
        | false -> master.Stop()
        logmessage "masterFarmer toggled."
            
        ()

File: regiondeployer/MasterSwitch.cs
using Microsoft.Azure.Management.Fluent;
using Microsoft.Azure.Management.AppService.Fluent;

namespace regiondeployer
{
    public class MasterSwitch
    {
        public const string MasterFarmerName = "chronictick";

        private IFunctionApp GetFirstMatch(Action<string> logMessage)
        {
            var functionApps = GetFunctionApps(logMessage);
            var output = functionApps.FirstOrDefault(currentApp => currentApp.Name.Contains(MasterFarmerName));
            return output;
        }

        public void ToggleMasterFarmer(bool shouldRun, Action<string> logMessage)
        {
            logMessage($"toggling master farmers operational state to {shouldRun}");

            var master = GetFirstMatch(logMessage);

            if (master == null)
            {
                throw new Exception($"no master farmer found. deploy that first. Its an azure function app whose name must contain {MasterFarmerName}");
            }
            else
            {
                if (shouldRun)
                {
                    master.Start();
                }
                else
                {
                    master.Stop();
                }
                logMessage("masterFarmer toggled.");
            }
        }
    }
}